### ISSUE TITLE
Defaults for responses

### ISSUE THREAD

**Issue by tquetano-r7** [camshaft/superagent-defaults#11]
More of question for help than an issue really, but was wondering ... is there a way to set any defaults upon response / end? I have a generic "strip prefix" function that needs to be called with all responses, and something like this would be handy:

superagent
    .on("end", (res) => {
        res.text = clearPrefix(res.text);
    });

**Comment by camshaft** [comment]
yeah that would be nice. I think you can accomplish that for now with something like:

```js
superagent
  .on('request', function (req) {
    req.on('end', function() {
      // do stuff here
    });
  });
```
